tests/tcp_stream.rs

```diff
@@ -36,21 +36,47 @@ fn is_send_and_sync() {
 #[test]
 #[cfg_attr(windows, ignore = "fails on Windows, see #1078")]
 fn tcp_stream_ipv4() {
-    smoke_test_tcp_stream(any_local_address());
+    let (poll, events) = init_with_poll();
+    let (handle, addr) = echo_listener(any_local_address(), 1);
+    let stream = TcpStream::connect(addr).unwrap();
```

Comment:
This is more setup then I would like. Why not change `smoke_test_tcp_stream` to accept an `FnOnce(SocketAddr) -> io::Result<TcpStream>` and use it like:

```rust
smoke_test_tcp_stream(any_local_address(), |addr| TcpStream::connect(addr));
```

This reduces the overhead of the setup, which is now in four functions.